### ISSUE TITLE
Login via Google doesn't work

### ISSUE THREAD

**Issue by Forinil** [davidbailey00/notion-linux#49]
I tried to log in via Google and it failed while redirecting me back form the browser with error that there is no application registered for URL _notion:/oauth2callback..._ 

I solved this by creating file _~/.local/share/applications/Notion.desktop_ with contents

```desktop
[Desktop Entry]
Type=Application
Name=Notion desktop application
Exec=notion-desktop %u
Icon=notion-desktop
StartupNotify=false
MimeType=x-scheme-handler/notion;
Type=Application
Categories=GNOME;GTK;Utility;

```

and then running

```shell
xdg-mime default Notion.desktop x-scheme-handler/notion
```

Later I found _/usr/share/applications/notion-desktop.desktop_ (which, curiosly, is still returned as a default handler by command `xdg-mime query default x-scheme-handler/notion`) and compared it to the one I created and it seems that missing `MimeType` declaration might be the issue.

**Comment by Forinil** [comment]
I forgot - I installed notion-desktop version 2.0.11-8_amd64 via apt and I'm running Kbuntu 20.04.2 LTS x86_64.

**Comment by SP-XD** [comment]
> I tried to log in via Google and it failed while redirecting me back form the browser with error that there is no application registered for URL _notion:/oauth2callback..._
> 
> I solved this by creating file _~/.local/share/applications/Notion.desktop_ with contents
> 
> ```desktop
> [Desktop Entry]
> Type=Application
> Name=Notion desktop application
> Exec=notion-desktop %u
> Icon=notion-desktop
> StartupNotify=false
> MimeType=x-scheme-handler/notion;
> Type=Application
> Categories=GNOME;GTK;Utility;
> ```
> 
> and then running
> 
> ```shell
> xdg-mime default Notion.desktop x-scheme-handler/notion
> ```
> 
> Later I found _/usr/share/applications/notion-desktop.desktop_ (which, curiosly, is still returned as a default handler by command `xdg-mime query default x-scheme-handler/notion`) and compared it to the one I created and it seems that missing `MimeType` declaration might be the issue.

Thanks It is working for me now, I am also in kubuntu 20.4 lts

**Comment by 3rgo** [comment]
Just installed today (version 2.0.11-8) on Ubuntu 20.04.2 LTS, and this fix did not work for me.
I tried creating the new desktop entry and editing the existing desktop entry (in /usr/share/applications) to add the MIME Type, nothing changes.
Any help appreciated ! Thanks !

**Comment by Forinil** [comment]
> Just installed today (version 2.0.11-8) on Ubuntu 20.04.2 LTS, and this fix did not work for me.
> 
> I tried creating the new desktop entry and editing the existing desktop entry (in /usr/share/applications) to add the MIME Type, nothing changes.
> 
> Any help appreciated ! Thanks !

Did you set the desktop entry with the MIME Type as a default handler using `xdg-mime` command as described in my original post?

**Comment by 3rgo** [comment]
> > Just installed today (version 2.0.11-8) on Ubuntu 20.04.2 LTS, and this fix did not work for me.
> > I tried creating the new desktop entry and editing the existing desktop entry (in /usr/share/applications) to add the MIME Type, nothing changes.
> > Any help appreciated ! Thanks !
> 
> Did you set the desktop entry with the MIME Type as a default handler using `xdg-mime` command as described in my original post?

Yes, and I also tried editing them to use the "default" notion-desktop app, nothing changed, no feedback in the app.
Maybe I did something wrong, I'm not very experienced in window manager configuration problems of this kind

**Comment by vtenfys** [comment]
I'm afraid I am no longer supporting this project, so I will not be looking into this issue. Please see the updated README for more information. I will leave the issue open, in case of a new maintainer who is able to look into it.

**Comment by 3rgo** [comment]
> I'm afraid I am no longer supporting this project, so I will not be looking into this issue. Please see the updated README for more information. I will leave the issue open, in case of a new maintainer who is able to look into it.

No worries, I've been using the web app in the mean time, I'll just keep using it :)

**Comment by SamuelFe** [comment]
> > Later I found _/usr/share/applications/notion-desktop.desktop_ (which, curiosly, is still returned as a default handler by command `xdg-mime query default x-scheme-handler/notion`) and compared it to the one I created and it seems that missing `MimeType` declaration might be the issue.
> 
> Thanks It is working for me now, I am also in kubuntu 20.4 lts

Hi there! Also in Kubuntu 20.04.2 LTS here. The solution worked!

### Quick way here:
```
sudo bash -c 'echo "MimeType=x-scheme-handler/notion;" >> /usr/share/applications/notion-desktop.desktop'
```

This will append the needed `MimeType=x-scheme-handler/notion;` into the original *notion-desktop.desktop* file.

*Obs.: this assumes that **notion-desktop.desktop** is in `/usr/share/applications/`*
